### PR TITLE
Skip imcompatible tests

### PR DESCRIPTION
**Pull request by jannyHou**
connect to strongloop/loopback-connector-cloudant#33
related to https://github.com/strongloop/loopback-connector-cloudant/pull/159

### CODE REVIEW

File: test/manipulation.test.js [hunk 1]
```diff
@@ -1822,6 +1825,11 @@ describe('manipulation', function() {
     });
 
     it('should allow delete(id) - success', function(done) {
+      // Suspect race condition happens somewhere
```
Comment (jannyHou) [inline test/manipulation.test.js]: I will remove the comments. Fixed now.

File: test/manipulation.test.js [hunk 2]
```diff
@@ -1837,6 +1845,7 @@ describe('manipulation', function() {
     });
 
     it('should allow delete(id) - fail', function(done) {
+      // Same reason, please check
```
Comment (jannyHou) [inline test/manipulation.test.js]: Ditto

File: test/manipulation.test.js [hunk 3]
```diff
@@ -1860,8 +1863,7 @@ describe('manipulation', function() {
       });
     });
 
-    bdd.itIf(connectorCapabilities.supportStrictDelete !== false, 'should allow delete(id) - ' +
-    'fail with error', function(done) {
+    it('should allow delete(id) - ' + 'fail with error', function(done) {
```
Comment (virkt25) [inline test/manipulation.test.js]: `supportStringDelete` flag shouldn't be removed along with the `cloudantCompatibility` flag.

File: test/manipulation.test.js [hunk 4]
```diff
@@ -2202,26 +2204,27 @@ describe('manipulation', function() {
...
-    'the where condition is not provided', function(done) {
-      filterHarry = connectorCapabilities.deleteWithOtherThanId === false ?
-        {id: idHarry} : {name: 'Harry Hoe'};
-      filterBrett = connectorCapabilities.deleteWithOtherThanId === false ?
-        {id: idBrett} : {name: 'Brett Boe'};
-      Person.update(filterHarry, function(err, info) {
-        if (err) return done(err);
-        info.should.have.property('count', 5);
-        Person.find({where: filterBrett}, function(err, people) {
+    bdd.itIf(connectorCapabilities.updateWithoutId !== false &&
```
Comment (jannyHou) [inline test/manipulation.test.js]: Hmm, hold on, IIRC this one should be a good test for cloudant.

File: test/manipulation.test.js (hunk 3 above)
Comment (jannyHou) [inline test/manipulation.test.js]: oops

**Pull request merged**